parsl/executors/errors.py

```diff
@@ -30,6 +30,24 @@ def __repr__(self):
                                                                 self.nodes_per_block)
 
 
+class UnsupportedFeatureError(ExecutorError):
+    """Error raised when attemping to use unsupported feature in an Executor"""
+
+    def __init__(self, feature, current_executor, target_executor):
+        self.feature = feature
+        self.current_executor = current_executor
+        self.target_executor = target_executor
```

Comment:
If we have a complex feature matrix, maintaining what target_executor(s) are supported might become annoying.
This is definitely useful to the user.